### PR TITLE
Make ensureVisible work with sliver-based viewports

### PR DESCRIPTION
**Pull request by abarth**
There appears to be an issue with children before the center widget.
I've filled a bug about that issue and will follow up in a later patch.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/rendering/sliver.dart [hunk 1]
```diff
@@ -1339,9 +1339,82 @@ abstract class RenderViewportBase2<ParentDataClass extends ContainerParentDataMi
...
+
+      assert(pivot.parent != null);
+      assert(pivot.parent != this);
+      assert(pivot != this);
+
+      final Matrix4 transform = targetBox.getTransformTo(pivot);
+      final Rect bounds = MatrixUtils.transformRect(transform, targetBox.paintBounds);
+
+      target = pivot;
+      assert(target.parent is RenderSliver);
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/sliver.dart]: maybe add TODO here mentioning that we should support other coordinate systems more generically

File: packages/flutter/lib/src/rendering/sliver.dart [hunk 2]
```diff
@@ -1693,6 +1769,31 @@ class RenderViewport2 extends RenderViewportBase2<SliverPhysicalContainerParentD
   }
 
   @override
+  double scrollOffsetOf(RenderSliver child, double scrollOffsetWithinChild) {
+    final GrowthDirection growthDirection = child.constraints.growthDirection;
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/sliver.dart]: assert that child.parent==this

**Comment by Hixie** [comment]
LGTM

**Pull request merged**